### ISSUE TITLE
Error when uploading RubyGems packages

### ISSUE THREAD

**Issue by deanpcmad** [go-gitea/gitea#19837]
### Description

When trying to upload a RubyGem package to Gitea (main version 1.17.0+dev-629-ge0273bad7), I get a `cannot unmarshal !!seq into string` error. 

```bash
gem push --host http://localhost:3000/api/packages/dean/rubygems mygem.gem

Pushing gem to http://localhost:3000/api/packages/dean/rubygems...
yaml: unmarshal errors:
  line 57: cannot unmarshal !!seq into string
```

Looking at the server logs, it shows this:

```
2022/05/29 11:58:23 ...rubygems/rubygems.go:24:apiError() [E] [629351cf] yaml: unmarshal errors:
	  line 57: cannot unmarshal !!seq into string
2022/05/29 11:58:23 [629351cf] router: completed POST /api/packages/dean/rubygems/api/v1/gems for 127.0.0.1:39502, 500 Internal Server Error in 13.3ms @ rubygems/rubygems.go:193(rubygems.UploadPackageFile)
```

When trying to reproduce the error on try.gitea.io, it doesn't show an error but when checking my account, no Gem has been uploaded so I'm not sure if packages are fully enabled on that system?

### Gitea Version

1.17.0+dev-629-ge0273bad7

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

2.34.1

### Operating System

Ubuntu 22.04

### How are you running Gitea?

Just running the binary in the command line 

### Database

SQLite

**Comment by deanpcmad** [comment]
I've just tried uploading a Docker image and that worked fine

**Comment by delvh** [comment]
> When trying to reproduce the error on try.gitea.io, it doesn't show an error but when checking my account, no Gem has been uploaded so I'm not sure if packages are fully enabled on that system?

Yeah, packages are disabled for try.gitea.io as far as I know.
For the rest, I don't really have an idea where the problem could be coming from.

**Comment by KN4CK3R** [comment]
Could you upload you gem file and/or project? I added only a basic marshal implementation to get it working. Maybe something is missing.

**Comment by deanpcmad** [comment]
As it's a private gem, I can't share the contents of it but I did have issues uploading the rails gem

**Comment by KN4CK3R** [comment]
I have tested rack and rails without problems:
```
gem push --host http://localhost:3000/api/packages/KN4CK3R/rubygems rails-7.0.3.gem     
Pushing gem to http://localhost:3000/api/packages/KN4CK3R/rubygems...
```
![grafik](https://user-images.githubusercontent.com/1666336/175719443-942ac286-0e76-4620-9d24-527523b4ce0b.png)

**Comment by deanpcmad** [comment]
Hmm, I'll give it another go then. What version are you using there?

**Comment by KN4CK3R** [comment]
Gitea is current master but there were no changes to the packages registry since 1.17-RC1.
```
gem -v
3.2.5
```

**Comment by deanpcmad** [comment]
Just tried it with another gem of mine, [twitchrb](https://github.com/deanpcmad/twitchrb/releases/tag/v1.0.0) and I get a similar error, but different lines:

```
yaml: unmarshal errors:
  line 29: cannot unmarshal !!seq into string
```

**Comment by KN4CK3R** [comment]
Thank you for the new example. Found the error this time and fixed it in #20470.

**Comment by DeanoCodes** [comment]
Just to confirm this is now fixed for a few Rubygems I've tested. Thanks!
